libcasr/src/js.rs

```diff
@@ -0,0 +1,281 @@
+        let Some(cap) = re.captures(stream) else {
+            return Err(Error::Casr(
+                "Couldn't find traceback in JS report".to_string(),
+            ));
+        };
+        let stacktrace = cap
+            .get(1)
+            .unwrap()
+            .as_str()
+            .split('\n')
+            .filter(|l| !l.is_empty())
+            .map(|l| l.to_string())
+            .collect::<Vec<String>>();
+
+        Ok(stacktrace)
+    }
+
+    fn parse_stacktrace_entry(entry: &str) -> Result<StacktraceEntry> {
+        let mut stentry = StacktraceEntry::default();
+        let re_full = Regex::new(
+            r"^(?:\s|\t)*at(?:\s|\t)+(.+?)(?:(?:\s|\t)+(\[as.*?\])?(?:\s|\t)*)\((.*)\)$",
+        )
+        .unwrap();
+        let re_without_pars =
+            Regex::new(r"^(?:\s|\t)*at(?:\s|\t)+(.+?)(?:(?:(?:\s|\t)+(\[as.*?\]))?(?:\s|\t)*)$")
+                .unwrap();
+
+        fn parse_location(
+            entry: &str,
+            stentry: &mut StacktraceEntry,
+            loc: &str,
+            can_be_func: bool,
+        ) -> Result<()> {
+            let mut debug: Vec<String> = loc.split(':').map(|s| s.to_string()).collect();
+            if debug.len() > 1 {
+                if debug.len() > 3 {
+                    let tail = debug[debug.len() - 2..].to_vec();
+                    debug = [debug[..debug.len() - 2].join(":")].to_vec();
+                    debug.append(&mut tail.to_vec());
+                }
+                stentry.debug.file = debug[0].to_string();
+                stentry.debug.line = if let Ok(line) = debug[1].parse::<u64>() {
+                    line
+                } else {
+                    return Err(Error::Casr(format!(
+                        "Couldn't parse line number {}. Entry: {entry}",
+                        debug[1]
+                    )));
+                };
+                if debug.len() == 3 {
+                    stentry.debug.column = if let Ok(column) = debug[2].parse::<u64>() {
+                        column
+                    } else {
+                        return Err(Error::Casr(format!(
+                            "Couldn't parse column number {}. Entry: {entry}",
+                            debug[2]
+                        )));
+                    }
+                }
+            } else if can_be_func {
+                if debug[0].ends_with(".js") || debug[0] == "native" {
+                    stentry.debug.file = debug[0].to_string();
+                } else {
+                    stentry.function = debug[0].to_string();
+                }
+            } else {
+                stentry.debug.file = debug[0].to_string();
+            }
+            Ok(())
+        }
+
+        fn parse_eval(entry: &str, stentry: &mut StacktraceEntry) -> Result<()> {
+            let re = Regex::new(
+                r"^(?:\s|\t)*at(?:\s|\t)+eval(?:\s|\t)+\((?:eval(?:\s|\t)+at(?:\s|\t)+(.*?)(?:\s|\t)+(?:(\[as.*?\])(?:\s|\t)+)?)(?:\((.*)\))(?:,(?:\s|\t)+(.*))\)$",
```

Comment:
Done, it's not necessary if we trim extra spaces in extract_stacktrace function  
Didn't replace (.*?) by (\S+) because function name may contain spaces  
Added more complicated processing because of possible variant that there is no function name, only location is present